app/models/file_depot_ftp.rb

```diff
@@ -93,12 +93,15 @@ def file_exists?(file_or_directory)
   private
 
   def create_directory_structure(directory_path)
-    Pathname.new(directory_path).descend do |path|
```

Comment:
> Also, if you have ideas on how to capture this behavior so we can automate tests
> to avoid regression, I'd love to hear any ideas.

I think the approach I had taken in the second commit of this PR can really help us in that regard.